// template-parser/template.js
module.exports = {
  profile: {
    selector: '.top-card-layout__card',
    fields: {
      full_name: 'h1',
      headline: '.top-card-layout__headline',
      location: 'h3 span:first-of-type',
      connections: 'h3 span:nth-of-type(2)',
      last_company: '.top-card__position-info span',
      last_education: '.top-card__education-info span',
      last_company_website: {
        selector: '.top-card__websites a',
        attribute: 'href'
      },
    }
  },
  educations: {
    is_multiple: true,
    selector: '.education li',
    fields: {
      name: 'h3',
      degree: 'h4 span:first-of-type',
      url: {
        selector: 'a',
        attribute: 'href'
      },
      field_of_study: 'h4 span:nth-of-type(2)',
      additional_info: 'h4 span:nth-of-type(3)',
      date1: '.date-range time:first-of-type',
      date2: '.date-range time:nth-of-type(2)',
      description: '.show-more-less-description p:last-of-type'
    }
  },
  about: {
    selector: '.summary',
    fields: {
      summary: 'p'
    }
  },
  positions: {
    is_multiple: true,
    selector: '.experience li:not(.experience-group-position)',
    fields: {
      title: 'h3',
      url: {
        selector: 'a',
        attribute: 'href'
      },
      company_name: 'h4',
      location: '.experience-item__location',
      description: '.show-more-less-description p:last-of-type',
      date1: '.date-range time:first-of-type',
      date2: '.date-range time:nth-of-type(2)',
      roles: {
        has_child: true,
        selector: 'li',
        fields: {
          title: 'h3',
          description: '.show-more-less-description p:last-of-type',
          date1: '.date-range time:first-of-type',
          date2: '.date-range time:nth-of-type(2)',
          location: '.experience-group-position__location'
        }
      }
    }
  },
  certifications: {
    is_multiple: true,
    selector: '.certifications li',
    fields: {
      name: 'h3',
      company: 'h4',
      url: {
        selector: 'h3 a',
        attribute: 'href'
      },
      company_url: {
        selector: 'h4 a',
        attribute: 'href'
      },
      date1: '.certifications__start-date time',
      date2: '.certifications__end-date time',
      certification_id: '.certifications__credential-id'
    }
  },
  people_also_view: {
    is_multiple: true,
    selector: '.browsemap li',
    fields: {
      full_name: 'h3',
      headline: 'h4',
      location: '.profile-result-card__location',
      url: {
        selector: 'a',
        attribute: 'href'
      }
    }
  },
  publications: {
    is_multiple: true,
    selector: '.publications li',
    fields: {
      name: 'h3',
      on: 'h4 span:first-of-type',
      date: '.date-range',
    }
  },
  courses: {
    is_multiple: true,
    selector: '.courses li',
    fields: {
      name: 'h3',
      date1: '.date-range time:first-of-type',
      date2: '.date-range time:nth-of-type(2)',
      description: '.show-more-less-description p:last-of-type',
      url: {
        selector: 'a',
        attribute: 'href'
      }
    }
  },
  groups: {
    is_multiple: true,
    selector: '.groups li',
    fields: {
      name: 'h3',
      description: '.show-more-less-description p:last-of-type',
      url: {
        selector: 'a',
        attribute: 'href'
      }
    }
  },
  projects: {
    is_multiple: true,
    selector: '.projects li:not(.face-pile__list-item)',
    fields: {
      name: 'h3',
      date1: '.date-range time:first-of-type',
      date2: '.date-range time:nth-of-type(2)',
      description: '.show-more-less-description p:last-of-type',
      url: {
        selector: 'a',
        attribute: 'href'
      },
      contributors: {
        has_child: true,
        selector: 'li',
        fields: {
          full_name: {
            selector: 'a, img',
            attribute: 'title, alt'
          },
          url: {
            selector: 'a',
            attribute: 'href'
          }
        }
      }
    }
  },
  awards: {
    is_multiple: true,
    selector: '.awards li',
    fields: {
      name: 'h3',
      company: 'h4',
      date: '.date-range',
      description: '.show-more-less-description p:last-of-type',
    }
  },
  skills: {
    is_multiple: true,
    selector: '.skills li',
    fields: {
      name: 'h3',
      subtitle: 'h4',
      description: '.show-more-less-description p:last-of-type',
    }
  },
  volunteer_experiences: {
    is_multiple: true,
    selector: '.volunteering li',
    fields: {
      name: 'h4',
      experience: 'h3',
      date1: '.date-range time:first-of-type',
      date2: '.date-range time:nth-of-type(2)',
      description: '.show-more-less-description p:last-of-type',
      field: '.volunteering__item--cause',
      url: {
        selector: 'a',
        attribute: 'href'
      }
    }
  },
  languages: {
    is_multiple: true,
    selector: '.languages li',
    fields: {
      name: 'h3',
      proficiency: 'h4',
      additional_info: '.result-card__meta '
    }
  },
  scores: {
    is_multiple: true,
    selector: '.scores li',
    fields: {
      name: 'h3',
      score: 'h4',
      date: '.date-range',
      description: '.show-more-less-description p:last-of-type',
    }
  },
  organizations: {
    is_multiple: true,
    selector: '.organizations li',
    fields: {
      name: 'h3',
      role: 'h4',
      date1: '.date-range time:first-of-type',
      date2: '.date-range time:nth-of-type(2)',
      description: '.show-more-less-description p:last-of-type',
      url: {
        selector: 'a',
        attribute: 'href'
      }
    }
  }
}
